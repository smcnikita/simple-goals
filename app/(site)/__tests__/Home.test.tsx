import { render, screen } from '@testing-library/react';
import { useTranslations } from 'next-intl';

import Home from '../page';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn(),
}));

const mockUseTranslations = useTranslations as jest.Mock;

describe('Home', () => {
  it('should renders Home unchanged', () => {
    mockUseTranslations.mockReturnValue((key: string) => {
      if (key === 'title') return 'Test Title';
      return key;
    });
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it('should renders the translated title', () => {
    mockUseTranslations.mockReturnValue((key: string) => {
      if (key === 'title') return 'Test Title';
      return key;
    });

    render(<Home />);

    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Test Title');
  });
});
