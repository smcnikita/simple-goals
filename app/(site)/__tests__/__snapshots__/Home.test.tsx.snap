// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should renders Home unchanged 1`] = `
<div>
  <div>
    <h1>
      Test Title
    </h1>
  </div>
</div>
`;
